Autohide: expose delay

pelatx opened the pull request:
Initial PR on the delay when we expose the outline (the autohide is activated).

Discussed in the issue #106.

Hirse commented on src/Autohide.js:
```diff
@@ -15,6 +15,7 @@ define(function (require, exports, module) {
     var isExposed = false;
     var sidebarPlusTransitionRemoved = false;
     var scrollTop;
+    var exposeDelay = prefs.get("delay");
```
This will only get the delay value once. In order to change it, one would have to restart/reload the extension.
Could you please use [`prefs.onChange`](https://github.com/Hirse/brackets-outline-list/blob/1ef0a91b288ed9c8be396c6ab8b46d0ca7a59ab5/src/Preferences.js#L126) instead to update the value on preference changes?

pelatx replied:
I made this change, but it is not working as expected. You will tell me if you find something that I should modify, please.

Hirse commented on src/Autohide.js:
```diff
@@ -104,6 +91,36 @@ define(function (require, exports, module) {
+            addSidebarPlusTransition();
+        });
+        $placeholder.hover(function () {
+            var timer = $(this).data("hover");
```
Please rename `"hover"` to something like `"expose-delay-timer"` (feel free to find something shorter).

Hirse commented on src/Preferences.js:
```diff
@@ -73,6 +73,10 @@ define(function Preferences(require, exports, module) {
                 description: Strings.PREF_AUTOHIDE_DESC
             }
+        }, {
+            id: "delay",
```
Please rename to `"autohideDelay"` (or equivalent) to indicate the connection to the autohide feature.

Hirse commented on src/Preferences.js:
```diff
@@ -73,6 +73,10 @@ define(function Preferences(require, exports, module) {
+        }, {
+            id: "delay",
+            type: "number",
+            value: 500
```
Please add name and description like for the other settings.

Hirse commented:
After trying it out for a bit, I feel like 500ms delay feels a bit slow, maybe I would go with 300ms. Additionally, we should add some immediate feedback on hover so the user knows there is a delay and doesn't feel like it's broken. Maybe highlighting (as in slightly lighter background) the placeholder?
Also see https://ux.stackexchange.com/a/119975 for the UX perspective.

pelatx commented:
It's not so hard about the names, it's that I'm a bit lazy. Your comments were expected. :wink:

I added the highlighting and changed what you said. Let me know if you find anything else.

pelatx commented:
OK @Hirse.

I had thought about implementing a range slider so that the user can change the delay. From an entry in the view menu.

I will do it to entertain me. But I'll show it to you because maybe you find it interesting to be added.

Hirse commented:
I would be happy to take a look whenever you have any features ready. 😁 
Thanks for your continued work on the extension.
